src/lib/utils/print_utils.hpp

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <ostream>
+
+namespace opossum {
+
+class Table;
+
+template <typename Node>
+using NodeGetChildrenFn = std::function<std::vector<std::shared_ptr<Node>>(const std::shared_ptr<Node>&)>;
+
+template <typename Node>
+using NodePrintFn = std::function<void(const std::shared_ptr<Node>&, std::ostream& stream)>;
+
+namespace detail {
+
+/**
+ *
+ * @param indentation   its size determines the indentation of a node, a true means a vertical line "|",
+ *                      a false means, a space " " should be used to increase the indentation
+ * @param id_by_node    used to determine whether a node was already printed and which id it had
+ * @param id_counter    used to generate ids for nodes
```

Comment:
Since we referenced the contribution guidelines several times today: `Maintain correct orthography and grammar. Comments should start with a capital letter, sentences should be finished with a full stop.`